src/python/WMCore/Services/Rucio/Rucio.py

```diff
@@ -360,7 +360,13 @@ def attachDIDs(self, rse, superDID, dids, scope='cms'):
         """
         if not isinstance(dids, list):
             dids = [dids]
-        dids = [{'scope': scope, 'name': did} for did in dids]
+        newDIDs = []
+        for did in dids:
+            if '#' in did:
```

Comment:
Is there any reason to implement this check for block here? I would simply write it as we discussed last week, e.g.:
```
dids = [{'scope': 'cms', 'name': did} for did in dids]
```
and add a comment that, if DIDs need to be attached against a different scope, further development is needed.

Reply:
I think it is correct check since block must be always be in CMS scope while datasets (rucio container may not). Since we should acknowledge given scope to this function I provided correct behavior, i.e. in input scope can be either cms or custom one and we should follow what is provided. Your suggestion will break this.

Reply:
We don't attach container to containers. So setting cms scope for those dids is fine.

Reply:
Alan, please scan Rucio.py code to see where attachDIDs is used, e.g. createBlock, createReplicas,  The later has specific scope input parameter, therefore I do not know how it will affect the other use-cases. I provided safest and correct way to follow the given scope and only make exception for dids which are CMS blocks.

Reply:
I really would like to avoid checking for a block by verifying the existence of the `#` symbol, which is a fragile logic. That's the reason I am suggesting to simply set it to `cms` instead.

createBlock --> attaches blocks
createReplicas --> attaches files/replicas. So hardwiring it to `cms` is as good as the current logic, but cleaner.

Reply:
I'll not argue anymore, but I made my point. Said that, the code now has changed to use your proposal and I added proper comment. Please refresh the PR to see last commit.